modules/externalipam/netbox/netbox_client.rb

```diff
@@ -0,0 +1,167 @@
+require 'yaml'
+require 'externalipam/externalipam'
+require 'externalipam/ipam_helper'
+require 'externalipam/ipam_validator'
+require 'externalipam/api_resource'
+require 'externalipam/ip_cache'
+
+module Proxy::Netbox
+  # Implementation class for External IPAM provider Netbox
+  class NetboxClient
+    include Proxy::Log
+    include Proxy::Ipam::IpamHelper
+    include Proxy::Ipam::IpamValidator
+
+    def initialize(conf)
+      @api_base = "#{conf[:url]}/api/"
+      @token = conf[:token]
+      @api_resource = Proxy::Ipam::ApiResource.new(api_base: @api_base, token: "Token #{@token}")
+      @ip_cache = Proxy::Ipam::IpCache.instance
+      @ip_cache.provider_name('netbox')
+    end
+
+    def get_ipam_subnet(cidr, group_name = nil)
+      if group_name.nil? || group_name.empty?
+        get_ipam_subnet_by_cidr(cidr)
+      else
+        group_id = get_group_id(group_name)
+        get_ipam_subnet_by_group(cidr, group_id)
+      end
+    end
+
+    def get_ipam_subnet_by_group(cidr, group_id)
+      params = { status: 'active', prefix: cidr, vrf_id: group_id }
+      response = @api_resource.get("ipam/prefixes/", params)
+      json_body = JSON.parse(response.body)
+      return nil if json_body['count'].zero?
+      subnet_from_result(json_body['results'][0])
+    end
+
+    def get_ipam_subnet_by_cidr(cidr)
+      params = { status: 'active', prefix: cidr }
+      response = @api_resource.get("ipam/prefixes/", params)
+      json_body = JSON.parse(response.body)
+      return nil if json_body['count'].zero?
+      subnet_from_result(json_body['results'][0])
+    end
+
+    def ipam_groups
+      response = @api_resource.get('ipam/vrfs/')
+      json_body = JSON.parse(response.body)
+      groups = []
+      return groups if json_body['count'].zero?
+      json_body['results'].each do |group|
+        groups.push({ name: group['name'], description: group['description'] })
+      end
+      groups
```

Comment:
Personally I'd write this as:
```suggestion
      return [] if json_body['count'].zero?
      json_body['results'].map do |group|
        { name: group['name'], description: group['description'] }
      end
```

And if `json_body['results']` is always an array, even if the count is 0 then the separate `return` statement isn't needed.